### DISCUSSION TITLE
I am having trouble using pyglider.

### DISCUSSION THREAD

**Discussion by riaoelerich** [c-proof/pyglider#146] [Q&A]
Hello everyone, 

I would like to use pyglider to process Slocum glider data. When I installed pyglider today I ran into a couple of issues that I wanted to raise here. In the following I will list my steps to ensure that I have not done a mistake (I am very new to python and Slocum gliders): 

1) I have installed the required packages and created a new environment as suggested on the website.
conda create -n gliderwork
conda activate gliderwork
conda install -c conda-forge pyglider

2) I downloaded the example data manually, as
import pyglider.example_data as pexamp
pexamp.get_example_data('./')
is not working. 

3)  Having the example data I wanted to run the example code to make sure that everything is working properly. The code is shown in the shown on the getting started : slocum (https://pyglider.readthedocs.io/en/latest/getting-started-slocum.html)

I wanted to run the first couple of lines to ensure the packages are imported properly. I received the following error message: 
Attribute Error: pyglider.slocum' has no attribute 'binary_to_timeseries

I sort of solved this issue by copying the code from pyglider.slocum from github and replacing the old code. 

I then received another error message that dbdreader is not defined. I realized that dbdreader was not installed with the packages (I think its important to transform the binary code from for example *.dbd files?) I have tried to install dbdreader separately, but it failed miserably on both my MAC and my Windows Laptop. 

I received the following error message: 
```
%%%%%
Collecting dbdreader
  Using cached dbdreader-0.4.11.tar.gz (768 kB)
  Preparing metadata (setup.py) ... done
Requirement already satisfied: numpy in ./opt/anaconda3/envs/gliderwork/lib/python3.11/site-packages (from dbdreader) (1.24.1)
Building wheels for collected packages: dbdreader
  Building wheel for dbdreader (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [15 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.macosx-10.9-x86_64-cpython-311
      creating build/lib.macosx-10.9-x86_64-cpython-311/dbdreader
      copying dbdreader/__init__.py -> build/lib.macosx-10.9-x86_64-cpython-311/dbdreader
      copying dbdreader/dbdreader.py -> build/lib.macosx-10.9-x86_64-cpython-311/dbdreader
      running build_ext
      building '_dbdreader' extension
      creating build/temp.macosx-10.9-x86_64-cpython-311
      creating build/temp.macosx-10.9-x86_64-cpython-311/extension
      clang -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /Users/riaoelerich/opt/anaconda3/envs/gliderwork/include -fPIC -O2 -isystem /Users/riaoelerich/opt/anaconda3/envs/gliderwork/include -Iextension/include -I/Users/riaoelerich/opt/anaconda3/envs/gliderwork/include/python3.11 -c extension/dbdreader.c -o build/temp.macosx-10.9-x86_64-cpython-311/extension/dbdreader.o
      xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for dbdreader
  Running setup.py clean for dbdreader
Failed to build dbdreader
Installing collected packages: dbdreader
  Running setup.py install for dbdreader ... error
  error: subprocess-exited-with-error
  
  × Running setup.py install for dbdreader did not run successfully.
  │ exit code: 1
  ╰─> [17 lines of output]
      running install
      /Users/riaoelerich/opt/anaconda3/envs/gliderwork/lib/python3.11/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_py
      creating build
      creating build/lib.macosx-10.9-x86_64-cpython-311
      creating build/lib.macosx-10.9-x86_64-cpython-311/dbdreader
      copying dbdreader/__init__.py -> build/lib.macosx-10.9-x86_64-cpython-311/dbdreader
      copying dbdreader/dbdreader.py -> build/lib.macosx-10.9-x86_64-cpython-311/dbdreader
      running build_ext
      building '_dbdreader' extension
      creating build/temp.macosx-10.9-x86_64-cpython-311
      creating build/temp.macosx-10.9-x86_64-cpython-311/extension
      clang -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /Users/riaoelerich/opt/anaconda3/envs/gliderwork/include -fPIC -O2 -isystem /Users/riaoelerich/opt/anaconda3/envs/gliderwork/include -Iextension/include -I/Users/riaoelerich/opt/anaconda3/envs/gliderwork/include/python3.11 -c extension/dbdreader.c -o build/temp.macosx-10.9-x86_64-cpython-311/extension/dbdreader.o
      xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> dbdreader

note: This is an issue with the package mentioned above, not pip.
```
%%%%

Sorry for this rather lengthy issue. I wonder whether some of you have encountered this? Or whether I did something wrong?

Best wishes,
Ria

**Comment by jklymak** [comment]
Hi Ria

To run dbdreader you will need a c compiler. So you need to install Xcode on your Mac or a c compiler on your windows machine (I don't have one so can't really help with this step) 

Note you do not need to use dbdreader.  There is a less robust two-step method to translate the files all in Python if you prefer.

**Comment by jklymak** [comment]
BTW< I've moved this to a discussion.  `xcode-select --install` should install Xcode for you.  Finally, I formatted your issue a bit to make the output a code block (three "```")

**Comment by riaoelerich** [comment]
Ok, I have now installed the dbdreader. Thanks for your help. 
I also redid all the steps and this time no issue with the Attribute Error: pyglider.slocum' has no attribute 'binary_to_timeseries either.
Looking forward to process my data with pyglider.

**Comment by callumrollo** [comment]
Thanks for raising this Ria. We should add info on the installations instructions about the need for a c compiler for dbdreader
